packages/asthma_diary_app/lib/feature/peak_flow/ui/component/peak_flow_chart.dart

```diff
@@ -175,6 +176,68 @@ class CurrentPeakFlowChartRangeText extends ConsumerWidget {
   }
 }
 
+/// 表示期間のセレクタ
+class _ChartRangeSelector extends ConsumerWidget {
+  const _ChartRangeSelector();
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final current = ref.watch(currentChartPeriodTypeNotifierProvider);
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        for (final ChartPeriodType period in ChartPeriodType.values) ...{
+          _ChartRangeSelectorButton(
+            item: period,
+            selected: period == current,
+            onPressed: () => ref
+                .watch(currentChartPeriodTypeNotifierProvider.notifier)
+                .updatePeriod(period),
+          ),
+        },
+      ],
+    );
+  }
+}
+
+/// 表示期間のボタン
+class _ChartRangeSelectorButton extends StatelessWidget {
+  const _ChartRangeSelectorButton({
+    required this.item,
+    required this.selected,
+    required this.onPressed,
+  });
+
+  final ChartPeriodType item;
+
+  /// 選択状態
+  final bool selected;
+
+  final VoidCallback onPressed;
+
+  @override
+  Widget build(BuildContext context) {
+    // 選択状態でボタン背景を変更
+    final textButtonStyle = selected
+        ? TextButton.styleFrom(
+            backgroundColor: context.primary,
+          )
+        : null;
+    // 選択状態で文字色を変更
+    final textStyle = selected
+        ? const TextStyle(
+            color: Colors.white,
+          )
+        : null;
+
+    return TextButton(
+      onPressed: onPressed,
+      style: textButtonStyle,
+      child: Text(item.displayName, style: textStyle),
+    );
```

Comment:
文字色は foregroundColor が使えます！あと、Whiteではなく primary の反対色 である onPrimary を使うと良い感じです！

```suggestion
    // 選択状態で背景色と文字色を変更
    final textButtonStyle = selected
        ? TextButton.styleFrom(
            backgroundColor: context.primary,
            foregroundColor: context.onPrimary,
          )
        : null;

    return TextButton(
      onPressed: onPressed,
      style: textButtonStyle,
      child: Text(item.displayName),
    );

```